mafiasi/vault/views.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+
+from django.contrib.auth.decorators import login_required
+from django.http.request import HttpRequest
+from django.http.response import HttpResponse, HttpResponseRedirect
+from django.template.response import TemplateResponse
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.views import View
+from django.views.decorators.csrf import csrf_protect
+
+from mafiasi.vault.vw_admin import VwUser, VwAdminClient
+
+
+@method_decorator(login_required, "dispatch")
+@method_decorator(csrf_protect, "dispatch")
+class IndexView(View):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.vw_client = VwAdminClient.init_from_settings()
+
+    def get_vw_user(self) -> Optional[VwUser]:
+        """
+        Retrieve the vaultwarden user corresponding to the currently logged in mafiasi user from the vaultwarden api
+        """
+        for user in self.vw_client.list_users():
+            if user.email == self.request.user.email:
+                return user
+        return None
+
+    def get(self, request: HttpRequest) -> HttpResponse:
+        vw_user = self.get_vw_user()
+        return TemplateResponse(
+            request,
+            "vault/index.html",
+            {
+                "vault_account_status": vw_user.status if vw_user else -1,
+                "vault_email": request.user.email,
```

Comment:
This should probably be real_email because it is shown to the user for confirmation and showing the cloak address to the user is not helpful.

Reply:
I thought about that too but email addresses are how users are identified in Vault. If you create a Vault organization you also invite users by their email address. I decided to use the cloak address intentionally so that users real email are not exposed to others.

This is definitely a tradeoff between privacy and being intuitive but I figured that since the email is shown on the vault page privacy is the way to go.

Reply:
> But this is not the data that goes to Vault, it is the information that the user sees in the template – and there I think it would still make sense to show the real email address.

I'm not sure I understand what you mean here.

Yes the email is what is shown to the user and what they log in with. This is intentional as it is also what is used by users to invite each other to vault organizations.

Reply:
After out talk, I have addressed this in aaa3c152328942a2ec2ab1a7747816d3a30fd3d5